src/ifcwrap-csharp/IfcOpenShell.Net/IfcOpenShell.Net/ArgumentResult.cs

```diff
@@ -0,0 +1,346 @@
+﻿using System;
+using System.Reflection.Emit;
+
+namespace IfcOpenShell
+{
+    public abstract class ArgumentResult
```

Comment:
Please add a class comment that answers the following questions:
- What does it do
- Why does it exist